### ISSUE TITLE
Contemplation about trasfering big portion of html extra files from Pulsar to Galaxy

### ISSUE THREAD

**Issue by martindemko** [galaxyproject/galaxy#18577]
Hi guys,
We serve a tool on our Galaxy that produces an HTML file with a bunch of extra files (could be over 10k files up to 50GB in total). These files are in the dataset_XYZ_files directory, and therefore, they are transferred from Pulsar to Galaxy (1st problem) and somehow post-processed on the Galaxy side (2nd problem) one by one, which is not good. We thought to reimplement the transfer so the _files directory could be zipped before and unzipped after the transfer. But that would not solve the 2nd problem. Another idea is to create a new data type, e.g., html_archive, which could wrap the extra files to the archive directly and also read them directly from the archive. Not sure how exactly yet. Maybe using some middleware in the way of IE. What do you think? Do you have an experience with so many files? Any suggestions about solving at least one of these problems without reimplementing half of Galaxy code is very welcomed. Thanks

**Comment by mvdbeek** [comment]
> (could be over 10k files up to 50GB in total).

I don't think this should be a problem per se, if you transfer them as an archive or not.

> they are transferred from Pulsar to Galaxy (1st problem)

Are you saying these files shouldn't be transferred ?

>  somehow post-processed on the Galaxy side (2nd problem) one by one, which is not good.

What exactly do you mean with post-processed (are you talking about the pulsar message ?), and what do you mean by one by one ?

**Comment by martindemko** [comment]
> > (could be over 10k files up to 50GB in total).
> 
> I don't think this should be a problem per se, if you transfer them as an archive or not.
> 
> > they are transferred from Pulsar to Galaxy (1st problem)
> 
> Are you saying these files shouldn't be transferred ?

Sorry for my misleading expression. They definitely should be transferred. What I meant is that all those files have to be transferred from Pulsar to Galaxy one by one, which creates unnecessary overhead and slows down network connection.

> 
> > somehow post-processed on the Galaxy side (2nd problem) one by one, which is not good.
> 
> What exactly do you mean with post-processed (are you talking about the pulsar message ?), and what do you mean by one by one ?

Again, sorry for my misleading expression. Here, I was talking about the Galaxy side. Specifically, the process of copying data one by one from new_file_path (typically some tmp directory) to file_path in case both of them reside on the same FS. If I understand it correctly, in this case, it could be done instantly by moving files.
